### ISSUE TITLE
[dotnet] latest STS version is stuck on 7.0.410

### ISSUE THREAD

**Issue by sliekens** [devcontainers/features#1213]
Hi,

It appears that the [latest.version (STS)](https://dotnetcli.azureedge.net/dotnet/Sdk/STS/latest.version) version is no longer being updated, it should report `9.0.101` but it's still on `7.0.410` three weeks after .NET 9 was released. As a result, the dotnet feature still defaults to installing the [latest.version (LTS)](https://dotnetcli.azureedge.net/dotnet/Sdk/LTS/latest.version) version (currently `8.0.403`) .

 Since the `latest.version` file above is more of a private API used by the dotnet-install script, maybe now is a good time to rewrite this code to use the [releases-index.json](https://dotnetcli.azureedge.net/dotnet/release-metadata/releases-index.json). More info: [README](https://github.com/dotnet/core/blob/main/release-notes/schemas/README.md).

**Comment by gauravsaini04** [comment]
Hi @sliekens 
The existing URL's have already been updated to point to the versions `9.0.101` for `STS` and `8.0.404` for `LTS`

**Comment by sliekens** [comment]
Ok interesting, was it coincidence? 😁

Regardless, I still think it's a good idea to use the releases index rather than the semi-private URL. I could draft a PR for it.

**Comment by gauravsaini04** [comment]
Yes sort of a coincidence, had been promptly updated by the dotnet team

The URL's that are present in the feature are maintained by dotnet team only

**Comment by sliekens** [comment]
@gauravsaini04 this just appeared on my feed: https://github.com/tonybaloney/Pyjion/pull/526
"new CDN link for dotnet"

It looks like the devcontainer feature might break after 15 Jan because of this change.

**Comment by richlander** [comment]
Yes, the files in this folder will need to change: https://github.com/devcontainers/features/tree/main/src/dotnet

Might it be time for the repo to snap a new copy of the install script? Or is it really necessary for this repo to keep its own copy with this logic?

https://github.com/dotnet/install-scripts/tree/main/src

**Comment by sliekens** [comment]
@richlander the dotnet-install.sh script in this repo is updated weekly via GitHub Actions ([update-dotnet-install-script.yml](https://github.com/devcontainers/features/blob/8e8e14f6fbb1f1c30e7794aff293960caba2945a/.github/workflows/update-dotnet-install-script.yml)), so that will get resolved as soon as the action runs and the resulting PR is merged.

The thing that won't resolve itself is the use of [latest.version (STS)](https://dotnetcli.azureedge.net/dotnet/Sdk/STS/latest.version) and [latest.version (LTS)](https://dotnetcli.azureedge.net/dotnet/Sdk/LTS/latest.version) to detect which channel has the absolute greatest version. I understand using these files for any purpose is not endorsed.

**Comment by sliekens** [comment]
I raised #1230 to stop using `azureedge.net`

**Comment by richlander** [comment]
I asked for the advice on the special files. Naturally, we're a bit short-staffed at the moment. We're going to publish guidance  on the various URL types, including the ones you've asked about.

**Comment by richlander** [comment]
My team got back to me. Those URLs are fine to use. They will be included in the guidance document.
